packages/flame_tiled/lib/src/renderable_tile_map.dart

```diff
@@ -194,11 +216,46 @@ class RenderableTiledMap {
 
   /// Render [batchesByLayer] that compose this tile map.
   void render(Canvas c) {
-    batchesByLayer.forEach((batchMap) {
-      batchMap.forEach((_, batch) => batch.render(c));
+    batchesByLayer.forEachIndexed((layerIndex, batchMap) {
+      batchMap.forEach((_tilesetImageFile, batch) {
+        final layer = map.layers[layerIndex];
+        c.save();
+
+        if (camera != null) {
+          final totalOffset = calculateParallaxOffset(camera!, layer);
+          c.translate(totalOffset.x, totalOffset.y);
+        }
+
+        // Paint with the layer's opacity
+        paint.color = paintColorByLayer[layerIndex];
+        batch.render(c, paint: paint);
+        c.restore();
+      });
     });
   }
 
+  /// Calculates the offset we need to apply to the canvas to compensate for
+  /// parallax positioning and scroll for the layer and the current camera position
+  /// https://doc.mapeditor.org/en/latest/manual/layers/#parallax-scrolling-factor
+  Vector2 calculateParallaxOffset(Camera camera, Layer layer) {
```

Comment:
This should use global private vector objects that it re-uses instead of creating this many new vector objects each tick.